scripts/simple-scylla-cluster-manager.py

```diff
@@ -0,0 +1,776 @@
+                raise RuntimeError("Cannot pass both --node-index and --all to node operation")
+            kwargs["all"] = False
+            for i in self._config["nodes"].keys():
+                kwargs["node_index"] = i
+                callback(**kwargs)
+            return True
+        elif not "node_index" in kwargs or not kwargs["node_index"] is None:
+            return False
+        else:
+            raise RuntimeError("Either --node-index or --all has to be passed to node operation")
+
+    def start_node(self, **kwargs):
+        if self._maybe_apply_all(self.start_node, **kwargs):
+            return
+        self._logger.info(f"start_node({kwargs})")
+
+        i = int(kwargs["node_index"])
+
+        if not i in self._config["nodes"]:
+            return self._make_fail(f"Failed to find node#{i}")
+
+        if i in self._nodes:
+            return self._make_success(f"Node#{i} is already running")
+
+        node_config = self._config["nodes"][i]
+
+        node_path = os.path.abspath(f"node{i}")
+        node_smp = self._get_node_config_item(i, "smp")
+        node_memory = self._get_node_config_item(i, "memory")
+        node_ip = node_config["ip"]
+
+        if self._nodes:
+            min_node_index = min(self._nodes.keys())
+            seed_node_ip = self._nodes[min_node_index]["ip"]
+        else:
+            seed_node_ip = node_ip
+
+        invoke_cmd = [os.path.join(node_path, "scylla"),
+                f"-c{node_smp}",
+                f"-m{node_memory}",
+                "--cluster-name", self._config['name'],
+                "--options-file", os.path.join(node_path, "scylla.yaml"),
+                "--developer-mode=1",
+                "--overprovisioned",
+                "--unsafe-bypass-fsync=1",
+                "--relaxed-dma",
+                "--listen-address", node_ip,
+                "--rpc-address", node_ip,
+                "--api-address", node_ip,
+                "--prometheus-address", node_ip,
+                "--seed-provider-parameters", f"seeds={seed_node_ip}",
+                "--data-file-directories", os.path.join(node_path, "data"),
+                "--commitlog-directory", os.path.join(node_path, "commitlog"),
+                "--hints-directory", os.path.join(node_path, "hints"),
+                "--view-hints-directory", os.path.join(node_path, "view_hints"),
+                "--api-ui-dir", os.path.join(node_path, "api-ui"),
+                "--api-doc-dir", os.path.join(node_path, "api-doc")]
+        invoke_cmd += self._get_node_config_item(i, "extra_args")
+
+        system_log_path = os.path.join(node_path, SCYLLA_LOG_FILE)
+        with open(system_log_path, "w") as system_log:
+            handle = subprocess.Popen(
+                    invoke_cmd,
+                    stdin = subprocess.DEVNULL,
+                    stdout = system_log,
+                    stderr = subprocess.STDOUT,
+                    cwd = os.path.abspath(node_path))
+            self._nodes[i] = {
```

Comment:
could use a `namedtuple` or `dataclass` if we don't really want to define a type for it. but better than a freestyle dict.

Reply:
Done in v2.